### PR TITLE
SKW-4168: create dataset backends

### PR DESCRIPTION
**Pull request by konvulse**
Adds generic method to create a backend, only valid backend types are accepted.

Adds convenience methods for all valid backend types.

No CLI support in this PR.

### CODE REVIEW

File: conduce/api.py [hunk 1]
```diff
@@ -756,6 +778,128 @@ def delete_transactions(dataset_id, **kwargs):
...
+    auto_process : boolean
+        Configure the backend to auto process transactions. Auto processing is enabled by default.
+        Set to False to disable auto processing.
+    **kwargs : key-value
+        See :py:func:`make_post_request` for more kwargs.
+    """
+    return _create_dataset_backend(dataset_id, DatasetBackends.tile, auto_process, None, **kwargs)
+
+
+def add_capped_tile_store(dataset_id, auto_process, minimum_temoral_level, maximum_temporal_level, **kwargs):
```
Comment (mbrennek) [inline conduce/api.py]: `minimum_temoral` level should be `minimum_temporal_level`

Comment (mbrennek) [reply conduce/api.py]: `minimum_temporal_level` and `maximum_temporal_level` are not documented in the parameters section.

Comment (konvulse) [reply conduce/api.py]: Wondering if `maximum_temporal_zoom` would be more clear.  Maybe `max_temporal_zoom_level`.

**Comment by konvulse** [comment]
Hmm, the capped tile store method doesn't seem to actually work.  Checking...

**Comment by konvulse** [comment]
Wow, I totally screwed up the config parameters.  max_temporal_level should be min_spatial_level.

File: conduce/api.py (hunk 1 above)
Comment (konvulse) [reply conduce/api.py]: Since I botched those parameters, I did wind up changing them to shorten to min/max and include the word zoom for clarity.

**Comment by mbrennek** [comment]
Going to merge this one to make reviewing the next two easier.  The Makefile can be fixed in a new PR.

**Pull request merged**